<?xml version="1.0"?>
<launch>
          
    <!-- Arguments that can be overridden from command line -->
    <arg name="frame_id" default="odom" />
    <arg name="use_rviz" default="true" /> 
    <arg name="point_size" default="0.1" />
    <arg name="line_width" default="0.05" />
    <arg name="arrow_scale" default="2" />
    <arg name="traj_display_length" default="400" /> <!-- Number of trajectory points to display -->

    <!-- Load parameters to parameter server -->
    <param name="frame_id" value="$(arg frame_id)" />
    
    <!-- Launch trajectory visualizer node -->
    <node name="traj_visualizer" pkg="traj_analyzer" type="traj_visualizer" output="screen">
        <param name="frame_id" value="$(arg frame_id)" />
        <param name="point_size" value="$(arg point_size)" />
        <param name="line_width" value="$(arg line_width)" />
        <param name="arrow_scale" value="$(arg arrow_scale)" />
        <param name="traj_display_length" value="$(arg traj_display_length)" />
    </node>
    
    <!-- Launch RViz with our configuration -->
    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find traj_analyzer)/rviz/traj_visualization.rviz" />
    </group>
    
    <!-- Remap topics if needed -->
    <!-- Example: <remap from="cmd_ref_trajectory" to="vehicle/trajectory/cmd_ref" /> -->
</launch>
